docs/docs/tutorial/part-4/index.mdx

```diff
@@ -0,0 +1,918 @@
+
+Conveniently, Gatsby has a powerful feature called the **data layer** that you can use to pull data into your site from anywhere. Want to keep your blog posts in WordPress, your store products in Shopify, and your user data in AirTable? No problem! With Gatsby's data layer, you can combine data from multiple sources, which lets you choose the best platform for each type of data.
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+Gatsby's data layer is powered by a technology called **GraphQL**. GraphQL is a query language with a special syntax that lets you ask for the data you need inside a component.
+
+In this Tutorial, we'll teach you all the GraphQL you'll need to know to build your first Gatsby site. Interested in learning more? [How To GraphQL](https://www.howtographql.com/) is a free tutorial that teaches you the fundamentals.
+
+</Announcement>
+
+In this part of the Tutorial, you'll learn about how to add data to Gatsby's data layer and how to pull that data into your React components.
+
+By the end of this part of the Tutorial, you will be able to:
+
+- Use **GraphiQL** to explore the data in the data layer and build your own GraphQL queries.
+- Use the **`useStaticQuery`** hook to pull data into a "building-block" component.
+- Use the **`gatsby-source-filesystem`** plugin to pull data into your site from your computer's filesystem.
+- Create a **page query** to pull data into a page component.
+
+## Meet Gatsby's GraphQL data layer
+
+Gatsby has its own [GraphQL](https://graphql.org/) data layer where it keeps all the data for your site. But how does it work under the hood?
+
+First, your data is stored in one or more **sources**. That source might be a folder on your computer's filesystem, a content management system (CMS) like WordPress, or a database. You might even have multiple data sources!
+
+How do you get from its source into the data layer? By adding special plugins to your site called **source plugins**. Each source plugin is designed to communicate with a specific source. When you build your site, each source plugin pulls data from its particular source and adds it to the GraphQL data layer for your site.
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+**Tip:** Curious what source plugins are in the [Plugin Library](/plugins/)? You can identify source plugins by their name: they typically start with `gatsby-source-`.
+
+For example, a few popular source plugins are `gatsby-source-filesystem` and `gatsby-source-contentful`.
+
+</Announcement>
+
+How do you get data back out of the data layer? You can write **GraphQL queries** inside of your components to pull out the data you want to use in your site. When you build your site, Gatsby will find all the GraphQL queries in your components, run them, and pass the resulting data into your component as a `data` prop.
```

Comment:
Technically this is only true for components/files inside `src/pages/` folder as you can do page queries there. Maybe make the last sentence more generic (without the `data` prop) so that it's true for both page queries and static queries

Reply:
Ah, good catch!

```suggestion
How do you get data back out of the data layer? You can write **GraphQL queries** inside of your components to pull out the data you want to use in your site. When you build your site, Gatsby will find all the GraphQL queries in your components, run them, and put the resulting data in your component.
```